File: Assets/Script/Dev/Mobile/RemixPreviewTest.lua
-- [ts]: RemixPreviewTest.ts
local ____exports = {} -- 1
local ____Dora = require("Dora") -- 1
local App = ____Dora.App -- 1
local Content = ____Dora.Content -- 1
local Size = ____Dora.Size -- 1
local sleep = ____Dora.sleep -- 1
local thread = ____Dora.thread -- 1
local ____Remix = require("Dev.Mobile.Remix") -- 2
local startMobileRemix = ____Remix.startMobileRemix -- 2
local resultPath = "/tmp/dora-mobile-remix-preview.result" -- 4
App.winSize = Size(390, 844) -- 6
thread(function() -- 7
	sleep(0.4) -- 8
	local host = startMobileRemix({ -- 9
		entry = {id = "mobile-remix-preview", title = "轨道花园", workDir = Content.assetPath, fileName = "Script/Dev/Mobile/FeedPreviewTest"}, -- 10
		onBack = function() return nil end, -- 16
		onPlay = function() return nil end -- 17
	}) -- 17
	sleep(0.8) -- 19
	local screenshot = App:saveScreenshot("/tmp/dora-mobile-remix-preview") -- 20
	sleep(0.3) -- 21
	Content:save( -- 22
		resultPath, -- 22
		(((((((((screenshot .. "\nvisual=") .. tostring(App.visualSize.width)) .. "x") .. tostring(App.visualSize.height)) .. " dpr=") .. tostring(App.devicePixelRatio)) .. " hostScale=") .. tostring(host.scaleX)) .. ",") .. tostring(host.scaleY) -- 22
	) -- 22
end) -- 7
return ____exports -- 7

File: Assets/Script/Dev/Mobile/RemixPreviewTest.ts
import { App, Content, Size, sleep, thread } from "Dora";
import { startMobileRemix } from "Dev/Mobile/Remix";

const resultPath = "/tmp/dora-mobile-remix-preview.result";

App.winSize = Size(390, 844);
thread(() => {
	sleep(0.4);
	const host = startMobileRemix({
		entry: {
			id: "mobile-remix-preview",
			title: "轨道花园",
			workDir: Content.assetPath,
			fileName: "Script/Dev/Mobile/FeedPreviewTest",
		},
		onBack: () => undefined,
		onPlay: () => undefined,
	});
	sleep(0.8);
	const screenshot = App.saveScreenshot("/tmp/dora-mobile-remix-preview");
	sleep(0.3);
	Content.save(resultPath, `${screenshot}\nvisual=${App.visualSize.width}x${App.visualSize.height} dpr=${App.devicePixelRatio} hostScale=${host.scaleX},${host.scaleY}`);
});
